### PR TITLE
fix(concertoForm): Properly update the cloned state on valueChange

### PR DESCRIPTION
**Pull request by mttrbrts**
Fixes an issue flagged in https://github.com/accordproject/concerto-ui/pull/58 which was prematurely merged during the migration to web-components

### CODE REVIEW

**Comment by irmerk** [comment]
Excellent! I think just needs to amend for the DCO signoff.

**Comment by mttrbrts** [comment]
Thanks @irmerk. Fixed

File: packages/concerto-ui-react/src/lib/concertoForm.js [hunk 1]
```diff
@@ -162,8 +163,9 @@ class ConcertoForm extends Component {
 
   onFieldValueChange(e, key) {
     const value = e.type === 'checkbox' ? e.checked : (e.value || e.target.value);
-    this.setState(set({ ...this.state.value}, key, value));
-    this.props.onValueChange(this.state.value);
+    const valueClone = set({ ...this.state.value}, key, value);
+    this.setState(valueClone);
```
Comment (DianaLease) [inline packages/concerto-ui-react/src/lib/concertoForm.js:167]: Just double-checking -- should this not be `this.setState({ value: valueClone })` ?

Comment (mttrbrts) [reply packages/concerto-ui-react/src/lib/concertoForm.js]: I miss my demo app! Thanks Diana, whatever would I do without you!

I've updated the story to make testing easier and I've also resolved along with some other issues that I found while testing.

**Pull request merged**